Python 3.7.3 (/usr/bin/python3)
>>> %Run get_battery_state.py
Checking RVR firmware versions...
Checking CMS firmware versions...
Firmware check done.
Battery percentage:  {'percentage': 89}
Voltage state:  {'state': 1}
Voltage states:  [unknown: 0, ok: 1, low: 2, critical: 3]
>>> %Run get_battery_state.py
Battery percentage:  {'percentage': 88}
Voltage state:  {'state': 1}
Voltage states:  [unknown: 0, ok: 1, low: 2, critical: 3]
>>> %Run get_battery_state.py
Traceback (most recent call last):
  File "/home/pi/sphero-sdk-raspberrypi-python/getting_started/observer/power/get_battery_state.py", line 22
    '{}: {}'.format(VoltageStates.literal garbage.name, VoltageStates.literal garbage.value),
                                                ^
SyntaxError: invalid syntax
>>> %Run get_battery_state.py
Traceback (most recent call last):
  File "/home/pi/sphero-sdk-raspberrypi-python/getting_started/observer/power/get_battery_state.py", line 22
    '{}: {}'.format(VoltageStates.literal garbage.name, VoltageStates.literal garbage.value),
                                                ^
SyntaxError: invalid syntax
>>> %Run get_battery_state.py
Traceback (most recent call last):
  File "/home/pi/sphero-sdk-raspberrypi-python/getting_started/observer/power/get_battery_state.py", line 23
    '{}: {}'.format(VoltageStates.super high.name, VoltageStates.super high.value),
                                           ^
SyntaxError: invalid syntax
>>> %Run get_battery_state.py
Traceback (most recent call last):
  File "/home/pi/sphero-sdk-raspberrypi-python/getting_started/observer/power/get_battery_state.py", line 24
    '{}: {}'.format(VoltageStates.you are bad at this game.name, VoltageStates.you are bad at this game.value)
                                        ^
SyntaxError: invalid syntax
>>> %Run get_battery_state.py
Battery percentage:  {'percentage': 87}
Voltage state:  {'state': 1}
Exception in thread serial_thread:
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/home/pi/.local/lib/python3.7/site-packages/sphero_sdk/observer/client/dal/serial_observer_port.py", line 49, in __serial_rw
    self.__read_bytes()
  File "/home/pi/.local/lib/python3.7/site-packages/sphero_sdk/observer/client/dal/serial_observer_port.py", line 56, in __read_bytes
    self.__parser.feed(data)
  File "/home/pi/.local/lib/python3.7/site-packages/sphero_sdk/observer/client/dal/observer_parser.py", line 19, in feed
    self.__read()
  File "/home/pi/.local/lib/python3.7/site-packages/sphero_sdk/observer/client/dal/observer_parser.py", line 47, in __read
    self.__dispatcher.handle_message(msg)
  File "/home/pi/.local/lib/python3.7/site-packages/sphero_sdk/observer/events/event_dispatcher.py", line 44, in handle_message
    self.__dispatch_event(handler, outputs, message)
  File "/home/pi/.local/lib/python3.7/site-packages/sphero_sdk/observer/events/event_dispatcher.py", line 57, in __dispatch_event
    handler(response_dictionary)
  File "/home/pi/sphero-sdk-raspberrypi-python/getting_started/observer/power/get_battery_state.py", line 21, in battery_voltage_handler
    '{}: {}'.format(VoltageStates.godly.name, VoltageStates.godly.value),
  File "/usr/lib/python3.7/enum.py", line 349, in __getattr__
    raise AttributeError(name) from None
AttributeError: godly

>>> %cd /home/pi/sphero-sdk-raspberrypi-python/getting_started/observer/leds
>>> %Run set_all_leds.py
>>> %cd /home/pi/dev/roundrock/desmond
>>> %Run set_all_leds.py
>>> %Run set_all_leds.py
>>> 'desmond'.upper()
'DESMOND'
>>> 'euowriyeroukuruyekygoiwytuwtoiw'.upper()
'EUOWRIYEROUKURUYEKYGOIWYTUWTOIW'
>>> 'lksuehtoiuehtvouesptiueoutirms;,euyrihutseyliurgynvseoiurykgduyhlfiseuyrvlbkeuvyfksluy;rbgouekuryvtwkleuyo'.upper()
'LKSUEHTOIUEHTVOUESPTIUEOUTIRMS;,EUYRIHUTSEYLIURGYNVSEOIURYKGDUYHLFISEUYRVLBKEUVYFKSLUY;RBGOUEKURYVTWKLEUYO'
>>> 'shrek'._class_
Traceback (most recent call last):
  File "<pyshell>", line 1, in <module>
AttributeError: 'str' object has no attribute '_class_'
>>> 'shrek'._class_str
Traceback (most recent call last):
  File "<pyshell>", line 1, in <module>
AttributeError: 'str' object has no attribute '_class_str'
>>> 'shrek'.__class__
<class 'str'>
>>> %cd /home/pi/Public/prog_pi_ed3
>>> %Run 05_01_converter.py
Convert inches to mm
converting 2 inches
50mm
>>> c2 = ScaleConverter('feet','yards', 0.333)
>>> print(str(c2.convert(3)) + c2.units_to)
0.9990000000000001yards
>>> 
